ls1intum/Artemis
# Show all editable exercise attributes in detail pages

### Checklist

### Motivation and Context


It solves Issue #1434 (Show all exercise attributes in detail page)

### Description

I added following attributes to all exercise detail pages (text, modeling, programming and file upload)
  - categories
  - difficulty
  - assessmentDueDate
  - if mode team is set, the team sizes are shown 

Furthermore:

- if no dates, level or category is set, there is a corresponding text shown
- fixed link to course page in detail page
- Aswell, rearranged the order for all details pages:
  - all elements of the superclass, then elements of the subclass
  - related items together
  - same order as in create / edit

Note: For the programming detail page I followed a slightly different order, as it would not make sense to have the short title e.g. below the superclass elements but rather below the actual title
### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Management
3. Go to Exercises
4. Create an Exercise of each type and Save
5. View Exercise Detail Page and check for all attributes

### Screenshots


**Programming Exercise Screenshots**
![image](https://user-images.githubusercontent.com/50466283/82804694-05986800-9e83-11ea-9c42-032a4d83cf7a.png)

**Modeling Exercise Screenshots**
![image](https://user-images.githubusercontent.com/50466283/82804884-5ad47980-9e83-11ea-900b-037c4f48574c.png)

**Text Exercise  Screenshots**

![image](https://user-images.githubusercontent.com/50466283/82804959-7d669280-9e83-11ea-81de-862cf3f87bd1.png)


**File Upload Exercise Screenshots**
![image](https://user-images.githubusercontent.com/50466283/82815573-bc074780-9e99-11ea-9ef1-b5328e32c1c1.png)
